### PR TITLE
Create shop order ui

### PR DESCRIPTION
**Pull request by nguyenminhduong**

### CODE REVIEW

File: app/src/main/java/com/example/duong/android_forder_01/data/model/OrderDetail.java [hunk 1]
```diff
@@ -37,4 +43,32 @@ public Product getProduct() {
...
+
+    public int getStatus() {
+        return mStatus;
+    }
+
+    public void setStatus(int status) {
+        mStatus = status;
+    }
+
+    public boolean getStatusPending() {
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/data/model/OrderDetail.java]: getStatusPending > isPendingStatus, isAcceptStatus...

File: app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    public CheckOrderAdapter(Context context, List<Orders> orders,
+                             CheckOrderContract.Presenter presenter) {
+        mContext = context;
+        mOrders = orders;
+        mListener = presenter;
+    }
+
+    @Override
+    public int getGroupCount() {
+        return mOrders.size();
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java]: check null

File: app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Override
+    public Object getGroup(int groupPosition) {
+        return mOrders.get(groupPosition);
+    }
+
+    @Override
+    public Object getChild(int groupPosition, int childPosition) {
+        return mOrders.get(groupPosition).getOrderDetails().get(childPosition);
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java]: getGroup(position) .getOrderDetails

File: app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Override
+    public long getGroupId(int groupPosition) {
+        return mOrders.get(groupPosition).getId();
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java]: getGroup(position) getID

File: app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Override
+    public long getChildId(int groupPosition, int childPosition) {
+        return mOrders.get(groupPosition).getOrderDetails().get(childPosition)
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java]: getChild.getId

File: app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    @Override
+    public View getChildView(int groupPosition, int childPosition, boolean isLastChild,
+                             View convertView, ViewGroup parent) {
+        if (convertView == null) {
+            LayoutInflater layoutInflater = (LayoutInflater) mContext
+                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            convertView = layoutInflater.inflate(R.layout.item_check_order, null);
+        }
+        ViewDataBinding mBinding = DataBindingUtil.bind(convertView);
+        mBinding.setVariable(BR.orderDetail,
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java]: Sử dụng hàm getChild(group, child)

File: app/src/main/java/com/example/duong/android_forder_01/ui/shopmanagementdetail/ordershop/checkorder/CheckOrderActionHandler.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    public void checkAll() {
+        mListener.checkAll();
+    }
+
+    public void checkOne(OrderDetail orderDetail) {
+        mListener.checkOne(orderDetail);
+    }
+
+    public void acceptAll() {
+        mListener.acceptAll();
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/shopmanagementdetail/ordershop/checkorder/CheckOrderActionHandler.java]: check null listenner

File: app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    public CheckOrderAdapter(Context context, List<Orders> orders,
+                             CheckOrderContract.Presenter presenter) {
+        mContext = context;
+        mOrders = orders;
+        mListener = presenter;
+    }
+
+    @Override
+    public int getGroupCount() {
+        if (mOrders == null) return 0;
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java]: Dùng toán tử 3 ngôi

File: app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        return mOrders.size();
+    }
+
+    @Override
+    public int getChildrenCount(int groupPosition) {
+        return mOrders.get(groupPosition).getOrderDetails().size();
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/adapter/CheckOrderAdapter.java]: getGroup.getOrderDetails

File: app/src/main/java/com/example/duong/android_forder_01/ui/shopmanagementdetail/ordershop/checkorder/CheckOrderFragment.java [hunk 10]
```diff
@@ -32,6 +42,41 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
...
+    }
+
+    @Override
+    public void shopListOrder(List<Orders> orders) {
+        mOrders.addAll(orders);
+        updateAdapter();
+    }
+
+    @Override
+    public void getListOrderError() {
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/duong/android_forder_01/ui/shopmanagementdetail/ordershop/checkorder/CheckOrderFragment.java]: TODO

Comment (nguyenminhduong) [inline app/src/main/java/com/example/duong/android_forder_01/ui/shopmanagementdetail/ordershop/checkorder/CheckOrderFragment.java]: fixed ạ

**Pull request merged**